website/static/js/filepage/index.js

```diff
@@ -157,6 +157,71 @@ var FileViewPage = {
         }, 1000);
 
         self.mfrIframeParent = $('#mfrIframeParent');
+
+        $.fn.editable.defaults.mode = 'inline';
+
+        if(self.canEdit) {
+            var $fileName = $('#fileName');
+            var conflict = 'warn';
+            var to = {
+                data: {
+                    nodeId: self.node.id,
+                    path: null,
+                    provider: self.file.provider,
+                }
+            };
+            var from = JSON.parse(JSON.stringify(to));
+            from.data.path = self.file.path;
+            $fileName.editable({
+                type: 'text',
+                send: 'always',
+                url: waterbutler.moveUrl(),
+                ajaxOptions:{
+                    beforeSend: $osf.setXHRAuthorization,
+                    headers: {
+                        'Content-Type': 'Application/json'
+                    },
+                },
+                params: function(params) {
+                    params = JSON.stringify({
+                        'rename': params.value,
+                        'conflict': conflict,
+                        'source': waterbutler.toJsonBlob(from),
+                        'destination': waterbutler.toJsonBlob(to),
+                    });
+                    return params;
+                },
+                validate: function(value) {
+                    if($.trim(value) === ''){
+                        $osf.growl('Error', 'Name cannot be empty.');
+                        return '   ';
+                    } else if(value.length > 100){
+                        $osf.growl('Error', 'Name should be less than 100 characters');
+                        return '   ';
+                    }
+                },
+                success: function(response, value) {
+                    window.location.reload();
+                },
+                error: function(response) {
+                    var msg = response.responseJSON.message;
+                    if (msg) {
+                        $osf.growl('Error', msg);
+                        return '      ';
+                    } else {
+                        msg = response.responseJSON.message_long;
+                        // Log unexpected error with Raven
+                        Raven.captureMessage('Error in renaming file', {
+                            url: waterbutler.moveUrl(),
+                            responseText: response.responseText,
+                            statusText: response.statusText
+                        });
+                        $osf.growl('Error', 'Error in renaming file. ' + msg);
```

Comment:
This growl box inserts unknown server-generated text directly into the page, and will treat the value of `message_long` as code. Consider using something like `$osf.htmlEscape` on the value of `msg`. 

If the error on the server contains both user-generated text *and* manually generated HTML, we can discuss some other strategies that may be helpful.